module/os/linux/zfs/zfs_vnops_os.c

```diff
@@ -398,11 +400,18 @@ zfs_zrele_async(znode_t *zp)
 	ASSERT(atomic_read(&ip->i_count) > 0);
 	ASSERT(os != NULL);
 
-	if (atomic_read(&ip->i_count) == 1)
+	/*
+	 * If decrementing the count would put us at 0, we can't do it inline
+	 * here, because that would be synchronous. Instead, dispatch an iput
+	 * to run later.
+	 *
+	 * For more information on the dangers of a synchronous iput, see the
+	 * header comment of this file.
```

Comment:
It would be nice if the section that talks about the dangers of synchronous iput mentioned the term `iput`, for easier searching.  (i.e. `zrele == iput`)